From 1eacd30724559749be5adeb31d763f44c3f221f9 Mon Sep 17 00:00:00 2001
From: Czarek Nakamoto <cyjan@mrcyjanek.net>
Date: Mon, 18 Nov 2024 10:57:37 -0500
Subject: [PATCH 13/14] include locale only when targeting WIN32


---
 src/wallet/api/wallet.cpp | 2 ++
 1 files changed, 2 insertions(+), 0 deletion(-)

diff --git a/src/wallet/api/wallet.cpp b/src/wallet/api/wallet.cpp
index d8fe108b4..e3e838b13 100644
--- a/src/wallet/api/wallet.cpp
+++ b/src/wallet/api/wallet.cpp
@@ -46,7 +46,9 @@
 #include <sstream>
 #include <unordered_map>
 
+#ifdef WIN32
 #include <boost/locale.hpp>
+#endif
 #include <boost/filesystem.hpp>
 #include "bc-ur/src/bc-ur.hpp"
 #if defined(HIDAPI_DUMMY) && !defined(HAVE_HIDAPI)
-- 
2.48.1

